### PR TITLE
Feature: Extend Hangfire Plugin with JobManager, Queue Support, and EF Health Check Integration

### PR DESCRIPTION
**Pull request by lukas-kd**
This PR enhances the Hangfire plugin by introducing a JobManager to orchestrate the various Hangfire job clients (BackgroundJobClient, RecurringJobManager, JobStorage) into a single, unified manager.

### Key Changes:

**JobManager**: Centralized manager for handling Hangfire job clients.
**Attribute Configuration**: Replaced the JobRegister and moved Timeout and TimeZone configurations to attribute-based setup for more flexible job management.
**Queue Support**: Added support for specifying job queues, allowing better control over job execution order and prioritization.

### Other improvements:
**Entity Framework Health Check**: Integrated automatic health check registration for Entity Framework, enhancing monitoring and resilience.
**WebHost Configurators**: Allows to implement configurators for plugins to allow more custom configuration.

### CODE REVIEW

File: PiBox.Hosting/WebHost/src/PiBox.Hosting.WebHost/Services/TypeImplementationResolver.cs [hunk 1]
```diff
@@ -36,6 +38,36 @@ private object GetArgument(Type instanceType, Type type)
...
+                {
+                    if (!isList)
+                        throw new NotSupportedException(
+                            $"For plugin configurators you must implement to take in an enumerable of the configurators. Configurator: {innerType.Name}");
+                    var args = _resolvedTypes.Select(x => x.Assembly).Distinct()
+                        .SelectMany(x => x.GetTypes())
+                        .Where(x => x.IsClass && !x.IsAbstract && x.IsAssignableTo(innerType))
+                        .Select(ResolveInstance)
+                        .ToList();
+                    if (!isList)
```
Comment (tst-sia) [inline PiBox.Hosting/WebHost/src/PiBox.Hosting.WebHost/Services/TypeImplementationResolver.cs:54]: code is unreachable as you're throwing an appropriate Exception above already, remove/redo

File: PiBox.Hosting/WebHost/test/PiBox.Hosting.WebHost.Tests/TypeImplementationResolverTests.cs [hunk 2]
```diff
@@ -83,6 +84,36 @@ public void CanResolvePluginConfigurations()
...
+            instance.Should().NotBeNull();
+            instance!.GetMessage().Should().Be("Hello World!");
+
+            var plugin = resolver.ResolveInstance(typeof(ConfiguratorPlugin)) as ConfiguratorPlugin;
+            plugin.Should().NotBeNull();
+            plugin!.Message.Should().Be("Hello World!");
+            plugin!.Message2.Should().Be("Hello World!");
+        }
+
+        internal class Configurator : IConfiguratorPluginConfigurator
```
Comment (tst-sia) [inline PiBox.Hosting/WebHost/test/PiBox.Hosting.WebHost.Tests/TypeImplementationResolverTests.cs:101]: can be private

File: PiBox.Hosting/WebHost/test/PiBox.Hosting.WebHost.Tests/TypeImplementationResolverTests.cs [hunk 3]
```diff
@@ -83,6 +84,36 @@ public void CanResolvePluginConfigurations()
...
+        {
+            public string GetMessage() => "Hello World!";
+        }
+
+        internal interface IConfiguratorPluginConfigurator : IPluginConfigurator
```
Comment (tst-sia) [inline PiBox.Hosting/WebHost/test/PiBox.Hosting.WebHost.Tests/TypeImplementationResolverTests.cs:106]: can be private

File: PiBox.Hosting/WebHost/test/PiBox.Hosting.WebHost.Tests/TypeImplementationResolverTests.cs [hunk 4]
```diff
@@ -83,6 +84,36 @@ public void CanResolvePluginConfigurations()
...
+        {
+            string GetMessage();
+        }
+
+        internal class ConfiguratorPlugin(IConfiguratorPluginConfigurator[] configurators, IList<IConfiguratorPluginConfigurator> configurators2) : IPluginActivateable
```
Comment (tst-sia) [inline PiBox.Hosting/WebHost/test/PiBox.Hosting.WebHost.Tests/TypeImplementationResolverTests.cs:111]: can be private

File: PiBox.Plugins/Jobs/Hangfire/src/PiBox.Plugins.Jobs.Hangfire/HangFirePlugin.cs [hunk 5]
```diff
@@ -13,49 +15,57 @@
...
             serviceCollection.AddHangfire(conf =>
                 {
                     conf.UseSerializerSettings(new JsonSerializerSettings());
                     if (_hangfireConfig.InMemory)
-                    {
                         conf.UseMemoryStorage();
-                    }
                     else
-                    {
                         conf.UsePostgreSqlStorage(opts => opts.UseNpgsqlConnection(_hangfireConfig.ConnectionString));
```
Comment (tst-sia) [inline PiBox.Plugins/Jobs/Hangfire/src/PiBox.Plugins.Jobs.Hangfire/HangFirePlugin.cs:45]: maybe we should make this configurable for other DB types someday

**Review by redbit86k — APPROVED** [review]

File: PiBox.Plugins/Jobs/Hangfire/src/PiBox.Plugins.Jobs.Hangfire/HangFirePlugin.cs (hunk 5 above)
Comment (lukas-kd) [reply PiBox.Plugins/Jobs/Hangfire/src/PiBox.Plugins.Jobs.Hangfire/HangFirePlugin.cs:45]: just added the configurator so you would be able to overwrite it from the outside of the plugin

**Comment by lukas-kd** [comment]
i've also adjusted the test method "GetInaccessibleValue" to be able to look for properties/fields on base type.
So we could easier test handlers on HttpClients and so on.

**Comment by tst-sia** [comment]
@lukas-kd We'll still need a couple days before we can test the changes as we're short on resources right now and priority doesn't open up much space. I'm certain there won't be that many comments or necessary alterations though.
I hope that's fine with you. 

In general we can talk about your expectations in regards to PR reviews so we can find a good middleground between testing on our side and not slowing you down/blocking any of your features.

File: PiBox.Plugins/Persistence/EntityFramework/src/PiBox.Plugins.Persistence.EntityFramework/EntityFrameworkPlugin.cs [hunk 6]
```diff
@@ -32,5 +34,17 @@ public void ConfigureApplication(IApplicationBuilder applicationBuilder)
         {
             DiagnosticListener.AllListeners.Subscribe(new DiagnosticObserver());
         }
+
+        public void ConfigureHealthChecks(IHealthChecksBuilder healthChecksBuilder)
```
Comment (tst-sia) [inline PiBox.Plugins/Persistence/EntityFramework/src/PiBox.Plugins.Persistence.EntityFramework/EntityFrameworkPlugin.cs:38]: This logic appears to be buggy as it looks for an IDbContext to register the EF health checks but inside the DI-Extensions class the IHealthChecksBuilder Extension method only accepts a TContext of type DbContext, not the interface. 
See:
https://github.com/sia-digital/pibox/blob/main/PiBox.Plugins/Persistence/EntityFramework/src/PiBox.Plugins.Persistence.EntityFramework/DependencyInjectionExtensions.cs#L29

This results in an exception as follows when starting a pibox dependent service that uses the EFPlugin: 
_System.Security.VerificationException: Method PiBox.Plugins.Persistence.EntityFramework.DependencyInjectionExtensions.AddEfContext: type argument 'Some.IExampleDbContext' violates the constraint of type parameter 'TContext'.
   at System.RuntimeMethodHandle.GetStubIfNeeded(RuntimeMethodHandleInternal method, RuntimeType declaringType, RuntimeType[] methodInstantiation)
   at System.Reflection.RuntimeMethodInfo.MakeGenericMethod(Type[] methodInstantiation)_

Consider changing the logic here or adding the interface as a valid type for the Extension method.

**Review by tst-sia — CHANGES_REQUESTED** [review]

File: PiBox.Plugins/Persistence/EntityFramework/src/PiBox.Plugins.Persistence.EntityFramework/EntityFrameworkPlugin.cs (hunk 6 above)
Comment (lukas-kd) [reply PiBox.Plugins/Persistence/EntityFramework/src/PiBox.Plugins.Persistence.EntityFramework/EntityFrameworkPlugin.cs:38]: the problem was, that it found the interface too.
I didnt filter for classes :dagger:

**Review by tst-sia — APPROVED** [review]

**Pull request merged**